### PR TITLE
Enable default Journal filter

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR add a filter option in journal controller at the client side, it's made by : 
- Adding a new service named `JournalFilteringService` 
- In this service we implement a function for filtering date in the 'yyyy-MM-dd' format

### CODE REVIEW

**Comment by sfount** [comment]
@jniles ▶️

File: client/src/partials/journal/journal.js [hunk 1]
```diff
@@ -29,39 +29,49 @@ JournalController.$inject = [
...
   // bind the transactions service to populate the grid component
   vm.gridOptions.data = Transactions.list.data;
 
-  // Initialise each of the journal utilities, providing them access to the journal
-  // configuration options
-  sorting = new Sorting(vm.gridOptions);
-  pagination = new Pagination(vm.gridOptions, Transactions.list.data);
-  grouping = new Grouping(vm.gridOptions);
+  // enable filter option
+  vm.gridOptions.enableFiltering = true;
```
Comment (jniles) [inline client/src/partials/journal/journal.js]: All other journal services import the `gridOptions` to set up specific options.  For example, [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalPagination.js#L97) in the pagination service configures external pagination.  In the same way, sorting sets up sorting in [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalSorting.js#L91).

This is a nice pattern developed by @sfount.  I propose that you refactor filtering to take in `gridOptions` like this:

``` js
pagination = new Pagination(vm.gridOptions, Transactions.list.data);
filtering  = new Filtering(vm.gridOptions);
```

and set up the `enableFiltering` property there, like the other services.

**Comment by jniles** [comment]
@mbayopanda - code LGTM.  One small refactor will make this code look like the other services and be more maintainable.  I'll merge it as soon as that lands.

Thanks!

**Comment by jniles** [comment]
@mbayopanda - the code looks good to me!

**Pull request merged**